### DISCUSSION TITLE
What can I do with emotion's theming that I can't do with a simple object or CSS vars?

### DISCUSSION THREAD

**Discussion by ivancuric** [emotion-js/emotion#2281] [General]
I'm currently building an app which will need theming support and I'm having a hard time understanding what benefits does emotion's theme API provide. It's basically a plain javascript object shared in a context or passed down as a prop.

Am I missing something?

**Comment by Andarist** [comment]
You can replicate Emotion's theming using CSS vars. Technically Emotion's theme can contain non-primitive values but that's not advised anyway.

When nesting ThemeProviders you can compute the inner theme based on the outer theme (which u cant do easily with CSS - at least not without extra rerender) but usually that's not needed either.

If you ask me - we'd like to drop having theme in the core but for now, we keep it to avoid churn in the community.

**Reply by ivancuric** [reply]
Thanks, good to know. I'll stick with CSS variables.
